python/snewpy/models/loaders.py

```diff
@@ -702,3 +702,83 @@ def get_initial_spectra(self, t, E, flavors=Flavor, interpolation='linear'):
                 raise ValueError('Unrecognized interpolation type "{}"'.format(interpolation))
 
         return initialspectra
+
+
+class Mori_2023(PinchedModel):
+    def __init__(self, filename, metadata={}):
+        """
+        Parameters
+        ----------
+        filename : str
+            Absolute or relative path to file prefix.
+        """
+        # PNS mass table, from Mori+ 2023.
+        mpns = { (0, 0):    1.78,
+                 (100, 2):  1.77,
+                 (100, 4):  1.76,
+                 (100, 10): 1.77,
+                 (100, 12): 1.77,
+                 (100, 14): 1.77,
+                 (100, 16): 1.77,
+                 (100, 20): 1.74,
+                 (200, 2):  1.77,
+                 (200, 4):  1.76,
+                 (200, 6):  1.75,
+                 (200, 8):  1.74,
+                 (200, 10): 1.73,
+                 (200, 20): 1.62 }
+
+        # Set up model metadata
+        self.axion_mass = metadata['Axion mass']
+        self.axion_coupling = metadata['Axion coupling']
+        self.progenitor_mass = metadata['Progenitor mass']
+
+        self.metadata = metadata
+
+        filebase = os.path.splitext(os.path.basename(filename))[0]
+        if 'std' in filebase:
+            am, ac = 0, 0
+        else:
+            am, ac = [int(_) for _ in filebase.split('_')[1:]]
+
+        self.pns_mass = mpns[(am,ac)] * u.Msun
+        self.metadata['PNS mass'] = self.pns_mass
```

Comment:
Here we're looking up the PNS mass from the given table, parsing the filename to get `am`,`ac` - to fill the metadata. Would it make more sense to do this lookup not in the loader, but in the `ccsn.Mori_2023` using the input parameters (or values from metadata)?

Reply:
@Sheshuk, this is a good point. I shifted the PNS table into the `Mori_2023` class in the `ccsn` module for better encapsulation and dealt with the complications resulting from adding units to all quantities.